save-core/src/commonMain/kotlin/org/cqfn/save/core/Save.kt

```diff
@@ -29,18 +49,75 @@ class Save(
      * Main entrypoint for SAVE framework. Discovers plugins and calls their execution.
      */
     fun performAnalysis() {
+        // FixMe: now we work only with the save.toml config and it's hierarchy, but we should work properly here with directories as well
+        // constructing the file path to the configuration file
+        val fullPathToConfig = saveProperties.testRootPath!!.toPath() / saveProperties.testConfigName!!.toPath()
         // get all toml configs in file system
-        val testConfig = ConfigDetector().configFromFile(saveProperties.testConfig)
+        val testConfig = ConfigDetector().configFromFile(fullPathToConfig)
+
+        val plugins: List<Plugin> = discoverPluginsAndUpdateTestConfig(testConfig)
+            // FixMe: Kirill should add merging of configs here
+            // ignoring the section with general configuration as it is not needed to create plugin processors
+            .pluginConfigs.filterNot { it is GeneralConfig }
+            .map {
+                when (it) {
+                    is FixPluginConfig -> FixPlugin(testConfig)
+                    is WarnPluginConfig -> WarnPlugin(testConfig)
+                    else -> throw PluginException("Unknown type <${it::class}> of plugin config was provided")
+                }
+            }
 
-        val plugins: List<Plugin> = emptyList()  // todo: discover plugins (from configuration blocks in TestSuiteConfig?)
         logInfo("Discovered plugins: $plugins")
         plugins.forEach { plugin ->
             logInfo("Execute plugin: ${plugin::class.simpleName}")
-            plugin.execute(testConfig).forEach(this::handleResult)
+            plugin.execute().forEach(this::handleResult)
             logInfo("${plugin::class.simpleName} successfully executed!")
         }
     }
 
+    private fun discoverPluginsAndUpdateTestConfig(testConfig: TestConfig): TestConfig {
+        val testConfigPath = testConfig.location.toString()
+        val parsedTomlConfig = TomlParser(testConfigPath).readAndParseFile()
+        parsedTomlConfig.getRealTomlTables().forEach {
+            // adding a fake file node to restore the structure and parse only the part of te toml
+            val fakeFileNode = TomlFile()
+            it.children.forEach {
+                fakeFileNode.appendChild(it)
+            }
+
+            val sectionName = it.name.toUpperCase()
+            val sectionPluginConfig: PluginConfig = when (val configName = TestConfigSections.valueOf(sectionName)) {
+                FIX -> createPluginConfig<FixPluginConfig>(testConfigPath, fakeFileNode, sectionName)
```

Comment:
I do not like this part, but don't know how to fix it. 
It could be unified with the following when-code now, but after @kgevorkyan will be able to merge configs - we won't be able to do this